azure-devops/run-ci-builds.ps1

```diff
@@ -3,31 +3,80 @@ param (
     [Parameter(Mandatory)] [string] $AzureDevOpsProjectName,
     [Parameter(Mandatory)] [string] $AzureDevOpsAccessToken,
     [Parameter(Mandatory)] [string] $SourceBranch,
-    [Parameter(Mandatory)] [string] $ToolVersions,
     [Parameter(Mandatory)] [UInt32] $DefinitionId,
-    [string] $SourceVersion
+    [Parameter(Mandatory)] [string] $SourceVersion,
+    [Parameter(Mandatory)] [string] $ManifestLink,
+    [Parameter(Mandatory)] [bool] $WaitForBuilds,
+    [string] $ToolVersions,
+    [UInt32] $RetryIntervalSec = 30,
+    [UInt32] $RetryCount = 20
 )
 
 Import-Module (Join-Path $PSScriptRoot "azure-devops-api.ps1")
 Import-Module (Join-Path $PSScriptRoot "build-info.ps1")
 
+function Get-ToolVersions {
+    param (
+        [Parameter(Mandatory)] [string] $ManifestLink,
+        [Parameter(Mandatory)] [UInt32] $RetryIntervalSec,
+        [Parameter(Mandatory)] [UInt32] $Retries,
+        [string] $ToolVersions
+    )
+    
+    [string[]] $versionsList = @()
+    if ($ToolVersions) {
+        $versionsList = $ToolVersions.Split(',')
+    } else {
+        Write-Host "Get the list of releases from $ManifestLink"
+        $releases = Invoke-RestMethod $ManifestLink -MaximumRetryCount $Retries -RetryIntervalSec $RetryIntervalSec
+        $versionsList = $releases.version
```

Comment:
Do we want to log versions that we have received?

Reply:
Yes, thanks